ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -131,15 +137,16 @@ object SqlLedgerReaderWriter {
     database.inWriteTransaction("Checking ledger ID at startup") { queries =>
       val providedLedgerId =
         initialLedgerId.getOrElse(Ref.LedgerString.assertFromString(UUID.randomUUID.toString))
-      val ledgerId = queries.updateOrRetrieveLedgerId(providedLedgerId)
-      if (initialLedgerId.exists(_ != ledgerId)) {
-        Future.failed(
-          new LedgerIdMismatchException(
-            domain.LedgerId(ledgerId),
-            domain.LedgerId(initialLedgerId.get),
-          ))
-      } else {
-        Future.successful(ledgerId)
+      Future.fromTry(queries.updateOrRetrieveLedgerId(providedLedgerId)).flatMap { ledgerId =>
```

Comment:
Considering the body of `flatMap` only constructs futures as `failed` and `successful` you can `flatMap` on the `Try` to avoid the context switch over the dispatcher.

Reply:
Good point. Done!